Week3/homework/step2-2.js

```diff
@@ -3,18 +3,30 @@
 function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
 
-  // Replace this comment and the next line with your code
-  console.log(startIndex, stopIndex, threeCallback, fiveCallback, numbers);
+  const count = stopIndex - startIndex + 1;
+  for (let i = 0; i < count; i++) {
+    numbers.push(startIndex);
+    startIndex++;
+  }
+
+  for (const item of numbers) {
+    if (item % 3 === 0 && item % 5 === 0) {
+      threeCallback(item);
+      fiveCallback(item);
+    } else if (item % 3 === 0) {
+      threeCallback(item);
+    } else if (item % 5 === 0) {
+      fiveCallback(item);
+    }
+  }
```

Comment:
Ah. I think you can improve the logic here a bit more. For example, if you think mathematically, a number which is divisible by both 3 and 5 will always be divisible by number 15, isn't it? Something like:

```
if (i % 15 === 0) {
  // call both three and five callbacks
} else if(i % 3 === 0) {
  // call three callbacks
} else if(i % 5 === 0) {
  // call five callbacks
}
```

Also, you need not have two separate loops for creating the `numbers` array and then for logging on console. You could do it in the same loop as well.

Also, you could rewrite your first loop as:

```
  for (let i = startIndex; i < stopIndex; i++) {
    numbers.push(i);
  }
``` 

🙂